dev-python/elasticsearch-py/elasticsearch-py-6.4.0.ebuild

```diff
@@ -0,0 +1,94 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{5,6} pypy )
+ES_VERSION="6.4.0"
+
+inherit distutils-r1
+
+MY_PN=${PN/-py/}
+DESCRIPTION="official Python low-level client for Elasticsearch"
+HOMEPAGE="https://github.com/elastic/elasticsearch-py"
+SRC_URI="https://github.com/elasticsearch/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	test? ( https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-${ES_VERSION}.tar.gz )"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="examples doc test"
+
+# tests fail in chroot
+# https://github.com/elastic/elasticsearch/issues/12018
+RESTRICT="test"
+
+RDEPEND=">=dev-python/urllib3-1.21.1[${PYTHON_USEDEP}]"
+
+DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
+	>=dev-python/sphinx-1.3.1-r1[${PYTHON_USEDEP}]
```

Comment:
Is this always needed, or only when building docs? I also suspect it's a BDEPEND, and that PYTHON_USEDEP isn't needed here.
setuptools is a BDEPEND and handled by distutils-r1.eclass (well you can add it if you want)

applies to 2nd ebuild as well :)

Reply:
Sphinx is needed for the man pages.

Reply:
Right, but its just *executed* once?

Reply:
So it means most of the packages under dev-python have it wrong? :)

Reply:
Well it's easy to test. Install sphinx with -python3.5 -python3.6 +python3.7, then do +python3.5 +python3.6 -python3.7 for elasticsearch.py. Does the build fail? Is the final list of installed files any different? 

I strongly believe sphinx is called _once_ during compilation, not everytime a new python interpr is used, so in my mind it'd be wrong to require sphinx to support every target. Not that it really matters though since sphinx isn't that big of package...

If you check closely how setuptools work, python3.6 is used to install stuff into /usr/lib64/python3.6/site-packages, python3.7 into /usr/lib64/python3.7/site-packages etc, but those files that sphinx creates is only installed once into /usr/share/man and /usr/share/doc not separated by python versions. As to "most of the packages under dev-python have it wrong? :)" yeah people still don't know how to use python eclasses properly, and address deps.

Anyway as I said, this isn't a blocker since sphinx is a small package, but I believe I'm right about it working without python_usedep.

Reply:
Yes, it makes sense, I'll update it with the next bump, thanks.